packages/bot-web-ui/src/components/quick-strategy/selects/contract-type.tsx

```diff
@@ -0,0 +1,112 @@
+                    setFieldValue?.(name, categories?.[0]?.value);
+                    setValue(name, categories?.[0]?.value);
+                    validateForm();
+                }
+            };
+            getContractTypes();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [symbol, tradetype]);
+
+    const handleChange = (value: string) => {
+        setFieldValue?.(name, value);
+        setValue(name, value);
+    };
+
+    const key = `qs-contract-type-${name}`;
+
+    return (
+        <div
+            className={classNames('qs__form__field no-top-border-radius no-top-spacing', {
+                'full-width': fullWidth,
+            })}
+        >
+            <Field name={name} key={key} id={key}>
+                {({ field }: FieldProps) => {
+                    const selected_item = list?.find(item => item.value === field.value);
+                    if (is_mobile) {
+                        return (
+                            <ul className='qs__form__field__list' data-testid='dt-qs-contract-types'>
+                                {list.map(item => {
+                                    const is_active = selected_item?.value === item.value;
+                                    return (
+                                        <li
+                                            key={item.value}
+                                            className={classNames('qs__form__field__list__item', {
+                                                'qs__form__field__list__item--active': is_active,
+                                            })}
+                                            onClick={() => {
+                                                handleChange(item.value);
+                                            }}
+                                        >
+                                            <Text size='xs' color='prominent' weight={is_active ? 'bold ' : 'normal'}>
+                                                {item.text}
+                                            </Text>
+                                        </li>
+                                    );
+                                })}
+                            </ul>
+                        );
+                    }
+                    return (
+                        <Autocomplete
+                            {...field}
+                            inputMode='none'
+                            data-testid='qs_autocomplete_contract_type'
+                            autoComplete='off'
+                            className='qs__select'
+                            value={selected_item?.text || ''}
+                            list_items={list}
+                            onItemSelection={(item: TItem) => {
+                                if (item?.value) {
+                                    handleChange((item as TContractTypesItem)?.value as string);
+                                }
```

Comment:
can we fix it like this:
<img width="982" alt="Screenshot 2023-11-02 at 11 40 46" src="https://github.com/binary-com/deriv-app/assets/103181650/b3ba119b-6fcc-4d22-bd16-2070e895142c">

```suggestion
                                const value = (item as TContractTypesItem)?.value as string;
                                if (value) {
                                    handleChange(value);
                                }
```

Reply:
maybe you can fix a similar thing in the file `contract-type.tsx` at the same time since type of TS issue is the same

Reply:
done 👍